docs/NaN_Full_release_docs.md

```diff
@@ -0,0 +1,249 @@
+    - `git push origin dev`
+    - if you are not an admin, create a pull request normally and ask someone with maintainer access to merge it
+- go to [Weblate's Maintenance tab](https://hosted.weblate.org/projects/newpipe/#repository)
+- press the *Update* button to update Weblate with the commit you just pushed on NewPipe's `dev` branch
+- **press the *Unlock*** button to allow translators to translate the changelog and possibly other components (**do not forget this step!**)
+- note that we had to do this process on NewPipe's `dev` branch because:
+    - Weblate's components are connected to NewPipe's `dev` branch and will update changes from there
+    - Weblate's git repo is not writable, so there is no way to push commits there manually
+
+### Creating the release branch
+
+- create a new branch starting from `dev`, named `release-X.X.X`, and switch to it
+    - `git checkout -b release-X.X.X`
+- edit the [`app/build.gradle`](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/build.gradle) file to update the extractor
+    - set the NewPipeExtractor dependency version to a suitable one (usually [the last commit in the NewPipeExtractor repo](https://github.com/TeamNewPipe/NewPipeExtractor/commits/dev))
+- commit the extractor update (if you used a specific version, specify `to VERSION` in the commit message)
+    - `git add app/build.gradle`
+    - `git commit -m "Update NewPipeExtractor"`
+- edit the [`app/build.gradle`](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/build.gradle) file to bump the release
+    - set `versionCode` to `NEW_VERSION_CODE`, i.e. increment the value by one as described in [Preliminary steps](#preliminary-steps)
+    - set `versionName` to `"X.X.X"`
+- commit the version bump (try to stick to the provided commit message template)
+    - `git add app/build.gradle`
+    - `git commit -m "Release vX.X.X (NEW_VERSION_CODE)"`
+- push the newly created branch to the NewPipe repo
+    - `git push upstream release-X.X.X`
+
+### Creating the Pull Request
+
+- create a Pull Request (PR) from the new branch you just pushed
+    - if you used the correct branch name you should be able to use this url, after changing the X.X.X: https://github.com/TeamNewPipe/NewPipe/pull/new/release-X.X.X
+- make sure the PR has `master` as the *base* branch and `release-X.X.X` as the *compare* branch
+- the PR title should be "Release vX.X.X (NEW_VERSION_CODE)"
+- remove all of the PR template and instead put these two lines in the description (the `ISSUE_NUMBER` will be replaced later):
+    ```
+    Do not report regressions here, but rather in the corresponding issue: #ISSUE_NUMBER
+    The changelog is also there.
+    ```
+- once you created the PR, note down its number (from now on called `PR_NUMBER`)
+- in case some issue should be closed when the release PR is merged, link them using the "Development" tab on the right, or add a "Fixes #...." in the PR description
```

Comment:
```suggestion
- in case some issue would be fixed when the release PR is merged, link them using the "Development" tab on the right, or add a "Fixes #...." in the PR description
```